parquet/read.go

```diff
@@ -67,8 +67,8 @@ func convertToSingleRowRecords(rec arrow.Record) []arrow.Record {
 	return records
 }
 
-// castExtensionColsToString casts extension columns to string.
-func castStringsToExtensions(rec arrow.Record, arrowSchema *arrow.Schema) (arrow.Record, error) {
+// castFromString casts extension columns to string.
+func castFromString(rec arrow.Record, arrowSchema *arrow.Schema) (arrow.Record, error) {
```

Comment:
Same issue the logic here is not recursive. Check out this example on how I did "zero copy" arrow recursive transformation - https://github.com/cloudquery/cloudquery/pull/10874

Reply:
read is only relevant to the types we have tests for though? If we need support for more types (by adding more tests) this should be adapted. Also, again, the default case of `cols[c] = col` would still work here.